module-system/sov-modules-macros/src/module_info.rs

```diff
@@ -208,11 +229,20 @@ fn make_init_state(
     })
 }
 
-fn make_init_module(field: &ModuleField) -> Result<proc_macro2::TokenStream, syn::Error> {
+fn make_init_module(
+    field: &ModuleField,
+    variant: ModuleType,
+) -> Result<proc_macro2::TokenStream, syn::Error> {
     let field_ident = &field.ident;
     let ty = &field.ty;
+    let trait_to_assert = match variant {
+        ModuleType::Standard => quote::quote! { ::sov_modules_api::Module },
+        ModuleType::Kernel => quote::quote! { ::sov_modules_api::KernelModule },
+    };
 
     Ok(quote::quote! {
+        // Ensure that the type implements "Module" or "KernelModule" at compile time
```

Comment:
Is it possible to check with try build if we get a suitable error message here?

Reply:
Yeah, that should be possible. I'll add this to the checklist 👍